sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MapStateImpl.scala

```diff
@@ -19,10 +19,27 @@ package org.apache.spark.sql.execution.streaming
 import org.apache.spark.internal.Logging
 import org.apache.spark.sql.Encoder
 import org.apache.spark.sql.catalyst.encoders.ExpressionEncoder
-import org.apache.spark.sql.execution.streaming.state.{PrefixKeyScanStateEncoderSpec, StateStore, StateStoreErrors, UnsafeRowPair}
+import org.apache.spark.sql.execution.streaming.TransformWithStateKeyValueRowSchema.{COMPOSITE_KEY_ROW_SCHEMA, VALUE_ROW_SCHEMA}
+import org.apache.spark.sql.execution.streaming.state.{ColumnFamilySchemaV1, PrefixKeyScanStateEncoderSpec, StateStore, StateStoreErrors, UnsafeRowPair}
 import org.apache.spark.sql.streaming.MapState
 import org.apache.spark.sql.types.{BinaryType, StructType}
 
+object MapStateImpl {
+  def columnFamilySchema[K, V](
```

Comment:
nit: could we make this a function for the Impl class below ?

Reply:
Was trying to do this - because the StateStore is absent from the driver, this would require us to pass None. I think the tradeoff of simplicity vs invasiveness of this change is not worth it, especially since it requires refactoring of the TTL code as well.

Reply:
Hmm - is it possible to move all these to a single singleton object then ?